src/tbp/monty/frameworks/models/evidence_matching/evidence_updater.py

```diff
@@ -0,0 +1,782 @@
+# Copyright 2025 Thousand Brains Project
+#
+# Copyright may exist in Contributors' modifications
+# and/or contributions to the work.
+#
+# Use of this source code is governed by the MIT
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Literal, Protocol, Tuple
+
+import numpy as np
+from scipy.spatial.transform import Rotation
+
+from tbp.monty.frameworks.models.evidence_matching.graph_memory import (
+    EvidenceGraphMemory,
+)
+from tbp.monty.frameworks.utils.evidence_matching import ChannelMapper
+from tbp.monty.frameworks.utils.graph_matching_utils import (
+    get_custom_distances,
+    get_initial_possible_poses,
+    get_relevant_curvature,
+)
+from tbp.monty.frameworks.utils.spatial_arithmetics import (
+    align_multiple_orthonormal_vectors,
+    get_angles_for_all_hypotheses,
+    get_more_directions_in_plane,
+    rotate_pose_dependent_features,
+)
+
+
+@dataclass
+class EvidenceUpdate:
+    evidence: np.ndarray
+    input_channel: str
+    locations: np.ndarray
+    poses: np.ndarray
```

Comment:
Does this refer to possible_locations and possible_poses? Can we make these variable names more explicit? We have locations and poses at many places in the code (sensed location and pose at each step, locations and poses stored in the buffer, hypothesized locations and poses on the object,...) so I think just using locations and poses as variable names can be confusing.

Reply:
Yes, the name change you suggested is much better.

I also added this commit, showing that the context can change as the update is passed. The contents of HypothesesUpdate go from "possible" to "new".

https://github.com/thousandbrainsproject/tbp.monty/pull/317/commits/90e4a34bab1429390fd47b9a22d09806813f22fa